app/controllers/users.js

```diff
@@ -16,16 +17,28 @@ const emailIsValid = email => email.match(validEmailPattern);
 
 const passwordIsValid = password => password.match(validPasswordPattern);
 
-exports.createUser = (req, res, next) => {
-  if (!obligatoryParametersWereReceived(req.body)) return next(errors.missingParameters);
+const validateUserCreation = parameters => {
+  if (!obligatoryParametersWereReceived(parameters)) return errors.missingParameters;
 
-  if (!emailIsValid(req.body.email)) return next(errors.invalidUserEmail);
+  if (!emailIsValid(parameters.email)) return errors.invalidUserEmail;
+
+  if (!passwordIsValid(parameters.password)) return errors.invalidUserPassword;
+
+  return null;
+};
 
-  if (!passwordIsValid(req.body.password)) return next(errors.invalidUserPassword);
+exports.createUser = (req, res, next) => {
+  const error = validateUserCreation(req.body);
+  if (error) return next(error);
+
+  const filter = {
+    email: req.body.email
+  };
 
   userService
-    .getUserByEmail(req.body.email)
-    .then(user => {
+    .getUsersByFilter(filter)
+    .then(users => {
```

Comment:
its fine like this, but keep in mind that its not rare to do a function to get just one user and one to get multiple users